nifi-commons/nifi-record/src/test/java/org/apache/nifi/serialization/record/ResultSetRecordSetTest.java

```diff
@@ -283,41 +544,235 @@ private ResultSet givenResultSetForArrayThrowsException(boolean featureSupported
+                RecordFieldType.DECIMAL, RecordFieldType.DOUBLE, RecordFieldType.FLOAT, RecordFieldType.INT, RecordFieldType.LONG, RecordFieldType.SHORT, RecordFieldType.STRING,
+                RecordFieldType.TIME, RecordFieldType.TIMESTAMP)
+                .map(RecordFieldType::getDataType)
+                .collect(Collectors.toList());
+        return RecordFieldType.CHOICE.getChoiceDataType(dataTypes);
+    }
+
+    private static class TestColumn {
+        private final int index; // Column indexing starts from 1, not 0.
+        private final String columnName;
+        private final int sqlType;
+        private final DataType recordFieldType;
+
+        public TestColumn(final int index, final String columnName, final int sqlType, final DataType recordFieldType) {
+            this.index = index;
+            this.columnName = columnName;
+            this.sqlType = sqlType;
+            this.recordFieldType = recordFieldType;
+        }
+
+        public int getIndex() {
+            return index;
+        }
+
+        public String getColumnName() {
+            return columnName;
+        }
+
+        public int getSqlType() {
+            return sqlType;
+        }
+
+        public DataType getRecordFieldType() {
+            return recordFieldType;
+        }
+    }
+
+    private static class SqlArrayDummy implements Array {
+
+        @Override
+        public String getBaseTypeName() throws SQLException {
+            return null;
+        }
+
+        @Override
+        public int getBaseType() throws SQLException {
+            return 0;
+        }
+
+        @Override
+        public Object getArray() throws SQLException {
+            return null;
+        }
+
+        @Override
+        public Object getArray(Map<String, Class<?>> map) throws SQLException {
+            return null;
+        }
+
+        @Override
+        public Object getArray(long index, int count) throws SQLException {
+            return null;
+        }
+
+        @Override
+        public Object getArray(long index, int count, Map<String, Class<?>> map) throws SQLException {
+            return null;
+        }
+
+        @Override
+        public ResultSet getResultSet() throws SQLException {
+            return null;
+        }
+
+        @Override
+        public ResultSet getResultSet(Map<String, Class<?>> map) throws SQLException {
+            return null;
+        }
+
+        @Override
+        public ResultSet getResultSet(long index, int count) throws SQLException {
+            return null;
+        }
+
+        @Override
+        public ResultSet getResultSet(long index, int count, Map<String, Class<?>> map) throws SQLException {
+            return null;
+        }
+
+        @Override
+        public void free() throws SQLException {
+
+        }
+    }
+
+    private static class BigDecimalDummy extends BigDecimal {
```

Comment:
Recommend renaming to `ResultBigDecimal`:
```suggestion
    private static class ResultBigDecimal extends BigDecimal {
```